src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -76,3 +77,44 @@ def test_set_wavelength_bad(inputs, msg):
     actual_args = argparse.Namespace(wavelength=inputs[0], anode_type=inputs[1])
     with pytest.raises(ValueError, match=re.escape(msg[0])):
         actual_args.wavelength = set_wavelength(actual_args)
+
+
+params5 = [
+    ([None], []),
+    (
+        [["facility=NSLS II", "beamline=28ID-2", "favorite color=blue"]],
+        [["facility", "NSLS II"], ["beamline", "28ID-2"], ["favorite color", "blue"]],
+    ),
+    ([["facility=NSLS II", "facility=NSLS III"]], [["facility", "NSLS III"]]),
+    ([["x=y=z"]], [["x", "y=z"]]),
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params5)
+def test_load_user_metadata(inputs, expected):
+    expected_parser = ArgumentParser()
+    expected_args = expected_parser.parse_args([])
+    for expected_pair in expected:
+        setattr(expected_args, expected_pair[0], expected_pair[1])
+
+    actual_parser = ArgumentParser()
```

Comment:
I still think we should use the actual parser, not create a parser called `actual_parser` (sorry if I wasn't clear).   It will start with something at the top of the file that looks like
```
from diffpy.labpdfproc.labpdfprocapp import get_args
```
and then you will pass these variables to `get_args`

Did you try that and couldn't make it work which is why you are doing it this way?  I know I suggested to do it this way at some point, but the advantage of doing it the other way if we can is that we actually test the api of hte program that the user will be using, because that parser sets some defaults and so our tests will be aware of those.

Reply:
Got it. I just tried get_args() and I think it needs mud to be a required input. So whenever I call actual_parser = get_args(), it'll always cause an error. Any idea how I might fix this?